terraform/lib/dependabot/terraform/registry_client.rb

```diff
@@ -46,7 +64,13 @@ def source(dependency:)
         # TODO: Implement service discovery for custom registries
         return unless hostname == PUBLIC_HOSTNAME
 
-        endpoint = "modules/#{dependency.name}/#{dependency.version}"
+        endpoint = if dependency.requirements.first[:source][:type] == "registry"
+                     "modules/#{dependency.name}/#{dependency.version}"
+                   elsif dependency.source[:type] == "provider"
+                     "providers/#{dependency.name}/#{dependency.version}"
+                   else
+                     raise "Invalid source type"
+                   end
```

Comment:
I kind wish we could write something like:

```ruby
[pluralize(dependency.source[:type]), dependency.name, dependency.version].join('/')
```

to collapse the conditional logic. I think this suggestion would be a bigger change so I'm not asking for it.